docs/docs/frontend-protocol.md

```diff
@@ -493,13 +496,21 @@ update
   ops: Op[]
   view-id: string
   pristine: bool
+  annotations: AnnotationSlice[]
 
 interface Op {
   op: "copy" | "skip" | "invalidate" | "update" | "ins"
   n: number  // number of lines affected
   lines?: Line[]  // only present when op is "update" or "ins"
   ln?: number // the logical number for this line; null if this line is a soft break
 }
+
+interface AnnotationSlice {
+  type: "find" | "selection" | ...
+  ranges: [[number, number, number, number]]  // start_line, start_col, end_line, end_col
+  payloads: [{}]    // can be any json object or value
+  n: number // number of ranges
+}
```

Comment:
This is good, but I think that we should also include some discussion of how annotations work, and how frontend authors should use this data. This might just be a link to the annotations RFC (which I guess we should merge)